docs: Validate database_observability.mysql  topic and consolidate examples (#6797)
grafana/alloy: docs/sources/collect/mysql-database-metrics.md

* Collect query performance data, explain plans, and lock information from MySQL.
* Monitor schema and table statistics.
* Forward database observability logs to Loki.
* Scrape MySQL metrics with Prometheus.

## Components used in this topic

* [`database_observability.mysql`][database_observability.mysql]
* [`prometheus.exporter.mysql`][prometheus.exporter.mysql]
* [`loki.relabel`][loki.relabel]
* [`discovery.relabel`][discovery.relabel]
* [`prometheus.scrape`][prometheus.scrape]
* [`prometheus.remote_write`][prometheus.remote_write]
* [`loki.write`][loki.write]

## Before you begin

* Ensure you have a running MySQL server with access credentials.
* Enable the MySQL `performance_schema` on your MySQL server.
  Refer to the [MySQL performance_schema documentation](https://dev.mysql.com/doc/refman/8.0/en/performance-schema.html) for setup instructions.
* Have write access to a Prometheus-compatible endpoint or Grafana Cloud for storing metrics.
* Have write access to a Loki-compatible endpoint or Grafana Cloud for storing logs.
* Be familiar with the concept of [Components][] in {{< param "PRODUCT_NAME" >}}.

## Collect MySQL database observability data

> The `database_observability.mysql` component connects to a MySQL database and collects performance schema data.
> This data includes query details, execution plans, and lock information forwarded as logs to Loki.
> The component also exports targets that can be scraped with Prometheus to collect MySQL metrics.

Reviewer: ```suggestion
The `database_observability.mysql` component connects to a MySQL database and collects performance telemetry:
* Loki logs for query details, execution plans, and lock information.
* Prometheus metrics for <>
```
Nit: it'd be nice to avoid general and ambigous terms such as "data" and to be more concise and straight to the point. People probably just wonder what telemetry they can collect. 

Also, logs and metrics are mentioned as sort of two different things, and it doesn't seem clear that these are the two signals that are being gathered.

In any case, it's probably best to mention this at the beginning of the page? There's already a list of bullet points there for similar things.

2. Add a `prometheus.exporter.mysql` component to export MySQL metrics.

   ```alloy
   prometheus.exporter.mysql "<LABEL>" {
     data_source_name  = "<MYSQL_DSN>"
     enable_collectors = ["perf_schema.eventsstatements"]
   }
   ```

   Replace the following:

   * _`<LABEL>`_: Must match the label used for the `database_observability.mysql` component.
   * _`<MYSQL_DSN>`_: The same MySQL Data Source Name as the `database_observability.mysql` component.

   For more information, refer to the [`prometheus.exporter.mysql`][prometheus.exporter.mysql] documentation.

3. Add a `loki.relabel` component to standardize labels on database logs.

   ```alloy
   loki.relabel "<LABEL>" {
     forward_to = [loki.write.logs_service.receiver]

     rule {
       target_label = "job"
       replacement  = "integrations/db-o11y"
     }

     rule {
       target_label = "instance"
       replacement  = "<LABEL>"
     }
   }
   ```

   Replace the following:

   * _`<LABEL>`_: Must match the label used for the `database_observability.mysql` component.

   For more information, refer to the [`loki.relabel`][loki.relabel] documentation.

> 4. Add a `discovery.relabel` component to standardize labels on Prometheus targets.

Reviewer: It's sad how we have to spend so much time describing relabelling and remote write. I agree that a user who wants to set up this component has to understand these concepts, but it feels like we're making the experience needlessly complicated. It would get especially repetitive if we have to mention this in every doc. Then a user who has to set up both MSSQL and MySQL has to read through or scroll through identical content. Is there a way to reduce this? E.g. we could just show a simple example which contains those components and include a link to a doc which explains how to do relabeling?

Author: Definitely. If we can identify the common "plumbing" components and document the concepts for them (vs the raw reference we have now) we'd have something to xref and it'd significantly reduce the middle content in any other task topic that used... loki.write, dicsovery.relabel and so on.
